fd/diff.go

```diff
@@ -19,61 +18,65 @@ type Point struct {
 	Coeff float64
 }
 
-// Method represents a finite difference formula that approximates
-// the derivative of order k of a function f at x as
-//  d^k f(x) ≅ (1 / h^k) * \sum_i Coeff_i * f(x + h * Loc_i),
-// where h is a small positive step.
-type Method struct {
+// Formula represents a finite difference formula on a regularly spaced grid
+// that approximates the derivative of order k of a function f at x as
+//  d^k f(x) ≈ (1 / Step^k) * \sum_i Coeff_i * f(x + Step * Loc_i).
+type Formula struct {
 	// Stencil is the set of sampling Points which are used to estimate the
 	// derivative. The locations will be scaled by Step and are relative to x.
 	Stencil []Point
 	Order   int     // The order of the approximated derivative.
-	Step    float64 // Default step size for the method.
+	Step    float64 // Default step size for the formula.
 }
 
 // Settings is the settings structure for computing finite differences.
 type Settings struct {
-	OriginKnown bool    // Flag that the value at the origin x is known
-	OriginValue float64 // Value at the origin (only used if OriginKnown is true)
-	Concurrent  bool    // Should the function calls be executed concurrently
-	Workers     int     // Maximum number of concurrent executions when evaluating concurrently
-	Method      Method  // Finite difference method to use
-}
-
-// DefaultSettings is a basic set of settings for computing finite differences.
-// Computes a central difference approximation for the first derivative
-// of the function.
-func DefaultSettings() *Settings {
-	return &Settings{
-		Method:  Central,
-		Workers: runtime.GOMAXPROCS(0),
-	}
+	// Formula is the finite difference formula used
+	// for approximating the derivative.
+	// Zero value will default to the Central formula.
+	Formula Formula
+	// Step is the distance between points of the stencil.
+	// If equal to 0, formula's default step will be used.
+	Step float64
+
+	OriginKnown bool    // Flag that the value at the origin x is known.
+	OriginValue float64 // Value at the origin (only used if OriginKnown is true).
+
+	Concurrent int // Maximum number of concurrent function evaluations.
 }
 
 // Derivative estimates the derivative of the function f at the given location.
-// The order of the derivative, sample locations, and other options are
-// specified by settings. If settings is nil, default settings will be used.
+// The finite difference formula, the step size, and other options are
+// specified by settings. If settings is nil, the first derivative will be
+// estimated using the Central formula and a suitable step size.
 func Derivative(f func(float64) float64, x float64, settings *Settings) float64 {
 	if settings == nil {
-		settings = DefaultSettings()
+		settings = &Settings{}
+	}
+	formula := settings.Formula
+	if formula.Stencil == nil {
```

Comment:
I think we should explicitly check that the formula is the zero value. Something like 
````
formula := settings.Formula
if formula == Formula{} {
    formula = Central
}else{
     if formula.Stencil == nil || formula.Step == 0 || formula.Order == 0 {
          panic("fd: bad formula")
    }
}